### DISCUSSION TITLE
Reload data after Next.js page change

### DISCUSSION THREAD

**Discussion by AkiHir0o** [swup/swup#921] [Help] [answered]
### Describe your problem 🧐

Hey, I've been trying for 3 days to understand and fix why, on the first load of my page, all the data is there, but when I switch pages, my JavaScript isn't executed, no useEffect or API requests.

I'm using Next.js with AppRouter. I have a layout.tsx file, and to apply all transitions on my website, I use a template.tsx where I use a useEffect to load Swup:

`use client

import { useEffect, useState } from 'react';

// Store
import { observer } from 'mobx-react-lite';
import swupStore from '@/store/swup';

import Swup from 'swup';
import './globals.css';

const Template = observer(({ children }: { children: React.ReactNode }) => {
  useEffect(() => {
    const swup = new Swup({
      containers: ['#swup'],
      cache: false,
    });

    // Initial URL setup
    swupStore.changePage(swup.currentPageUrl);

    const handleContentReplaced = () => {
      swupStore.changePage(swup.currentPageUrl);
    };

    const handlePageView = (visit: any) => {
      swupStore.changePage(visit.to.url);
    };

    swup.hooks.on('content:replace', handleContentReplaced);
    swup.hooks.on('page:view', handlePageView);

    return () => {
      swup.hooks.off('content:replace', handleContentReplaced);
      swup.hooks.off('page:view', handlePageView);
      swup.destroy();
    };
  }, []);

  return (
    <main id="swup" className="transition-fade h-full">
      {children}
    </main>
  );
});

export default Template;
`

### What is your question? 🙋‍♀️

How can I load my dynamic data when I change pages ?

If you want to test, try switching to the About page and Links page.

Thanks for your helps

### URL to minimal reproduction 🔗

https://kibo-website.vercel.app/

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by hirasso** [comment, marked as answer]
Hi there! Swup is usually being used as an *alternative* rather than a companion to JS frameworks like next.js with client-side routing. I don't have any experience with next.js, but is there some main script that needs to be initialized? If so, you could make use of swup's hook system to re-initialize your next.js scripts on every page load, like so:

```js
swup.hooks.on("page:view", (visit) => {
  // re-initialize your app 🤷‍♂️
});
```
